cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -233,30 +234,30 @@ public void onClose(KubernetesClientException e) {
     }
 
     /**
-     * Searches through the Conditions in the supplied status and returns the status of the Condition with type
-     * {@link KafkaRebalanceStatus#REBALANCE_STATUS_CONDITION_TYPE}.
+     * Searches through the Conditions in the supplied status and returns the type of the Condition with status
+     * {@link KafkaRebalanceStatus#REBALANCE_STATUS_CONDITION_STATUS}.
      *
      * @param status The status instance whose conditions will be searched.
      * @return The status of the rebalance condition.
      */
-    private String rebalanceStateConditionStatus(KafkaRebalanceStatus status) {
+    private String rebalanceStateConditionType(KafkaRebalanceStatus status) {
         Condition rebalanceStateCondition = rebalanceStateCondition(status);
-        return rebalanceStateCondition != null ? rebalanceStateCondition.getStatus() : null;
+        return rebalanceStateCondition != null ? rebalanceStateCondition.getType() : null;
     }
 
     private Future<KafkaRebalance> updateStatus(KafkaRebalance kafkaRebalance,
                                                 KafkaRebalanceStatus desiredStatus,
                                                 Throwable e) {
         // leaving the current status when the desired one is null
         if (desiredStatus != null) {
-            String rebalanceStatusString = rebalanceStateConditionStatus(desiredStatus);
+            String rebalanceTypeString = rebalanceStateConditionType(desiredStatus);
 
             if (e != null) {
                 StatusUtils.setStatusConditionAndObservedGeneration(kafkaRebalance, desiredStatus,
-                        KafkaRebalanceStatus.REBALANCE_STATUS_CONDITION_TYPE, State.NotReady.toString(), e);
-            } else if (rebalanceStatusString != null) {
+                        State.NotReady.toString(), KafkaRebalanceStatus.REBALANCE_STATUS_CONDITION_STATUS, e);
+            } else if (rebalanceTypeString != null) {
                 StatusUtils.setStatusConditionAndObservedGeneration(kafkaRebalance, desiredStatus,
-                        KafkaRebalanceStatus.REBALANCE_STATUS_CONDITION_TYPE, rebalanceStatusString);
+                        rebalanceTypeString, KafkaRebalanceStatus.REBALANCE_STATUS_CONDITION_STATUS);
```

Comment:
you are using this overload ...

```java
StatusUtils.setStatusConditionAndObservedGeneration(R resource, S status, String type, String conditionStatus, Throwable error)
```

where `conditionStatus` is actually always "True".
Maybe it could be useful having another overload without the `conditionStatus` where it's set to "True" implicitly.
Something like we already have:

```java
public static <R extends CustomResource, S extends Status> void setStatusConditionAndObservedGeneration(R resource, S status, Throwable error) {
        setStatusConditionAndObservedGeneration(resource, status, error == null ? "Ready" : "NotReady", "True", error);
    }
```

but with the additional `type` parameter.